### DISCUSSION TITLE
Building glitch - Format String Issue - Data argument not used by format string

### DISCUSSION THREAD

**Discussion by nodevnul** [ExistentialAudio/BlackHole#327] [Q&A]
Hi Devin.

I have followed your wiki to build seperate instances and get the following issue reporting shown below. In your wiki you write:

In BlackHole.h change the following UID to a different string.
#define    kBox_UID                   "BlackHole_UID"
#define    kDevice_UID              "BlackHole_UID"
#define    kDevice_ModelUID    "BlackHole_ModelUID"

where it is not quite clear which string to change:

a) Everything between the quotation marks ?
b) Only the part "UID" ?

I changed between the quotation marks as shown in the screenshots.

You might find this a dull question, but I am am new to this building stuff hence longing for precise information.

Thanks for listening.

Showing Recent Issues
/Volumes/Production/Development/BlackHole CoreAudio-Driver/BlackHole/BlackHole.c:912:111: Data argument not used by format string

/Volumes/Production/Development/BlackHole CoreAudio-Driver/BlackHole/BlackHole.c:955:114: Data argument not used by format string

/Volumes/Production/Development/BlackHole CoreAudio-Driver/BlackHole/BlackHole.c:1326:85: Data argument not used by format string

/Volumes/Production/Development/BlackHole CoreAudio-Driver/BlackHole/BlackHole.c:1907:88: Data argument not used by format string

/Volumes/Production/Development/BlackHole CoreAudio-Driver/BlackHole/BlackHole.c:1916:93: Data argument not used by format string

![Bildschirmfoto 2021-03-01 um 17 26 27](https://user-images.githubusercontent.com/69191814/109527480-fc909300-7ab3-11eb-889a-85e5ff70312c.png)
![Bildschirmfoto 2021-03-01 um 17 26 36](https://user-images.githubusercontent.com/69191814/109527492-00241a00-7ab4-11eb-8128-22bc3e260b6f.png)

**Comment by devinroth** [comment]
It’s just a compiler warning. Safe to ignore.

I have BlackHole to dynamically change the names depending on the number of channels.

That’s what the “%i” means.

**Reply by nodevnul** [reply]
Ah ok.

Yes I noticed the "wildcard" %ich and checked what the naming would be in Audio-MIDI-Setup.

But I wonder: Where is it defined that the

%ich

knows to insert the value of

#define                             NUMBER_OF_CHANNELS                  2

I could not find it.

Maybe you could add the formatstring warning to the wiki for newbees.

Is this format string thihg something that can be "fixed" easily - although not functionally restricting ?

**Comment by devinroth** [comment]
It says the exact location in the errors you pasted.

Yes you can take it out. But why? It literally means nothing. It’s just a reminder.

**Reply by nodevnul** [reply]
Ok, I saw the macro references, I meant:

Set %ich = NUMBER_OF_CHANNELS

**Comment by devinroth** [comment]
The UIDs are defined in the header. The channel number is inserted at those locations previously referenced.

There’s nothing more to it.

**Reply by nodevnul** [reply]
Ok - Thanks.
